jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1011,6 +1011,27 @@ void java_object_factoryt::gen_nondet_init(
     assign.add_source_location() = location;
 
     assignments.add(assign);
+    // add assumes to obey numerical limits (given via
+    // java-assume-inputs-interval)
+    if(type != java_boolean_type() && type != java_char_type())
```

Comment:
Why is `java_char_type()` also special cased? It would seem that intervals still make sense for that type?

Reply:
it makes sense but we would probably want a different one, for instance if we use [0:10] for integers it would only give non-printable characters

Reply:
I think that wouldn't be very intuitive for the user (definitely not for me :slightly_smiling_face: ). We have `--string-input-value` option which accepts a list of strings to use during test generation (it's not exactly the same though as what you meant).

Reply:
A little forward visibility - a forthcoming feature will bound `java_char_type` when `--string-printable` is used (using the same range of characters).